include/gzip/compress.hpp

```diff
@@ -1,8 +1,10 @@
-#include <iostream>
```

Comment:
`<iostream>` was uneeded, so I removed. But to keep the code compiling I needed to add `<string>` (turns out <string> was being included by <iostream>).